### PR TITLE
clientv3: add send created notification

### PR DESCRIPTION
**Pull request by xiang90**
For #6012

### CODE REVIEW

File: clientv3/watch.go [hunk 1]
```diff
@@ -49,6 +49,11 @@ type Watcher interface {
 	Close() error
 }
 
+type CreatedNotifier interface {
```
Comment (heyitsanthony) [inline clientv3/watch.go]: why not `WithCreateNotify()`? This seems very similar to `WithProgressNotify()`

Comment (xiang90) [inline clientv3/watch.go]: Not really. This is for entire watch client, not just one watcher. Right now, we do not have mechinism to propagate per watcher configuration. Things like ProgressNotify are set directly to the server side, not controlled at client side. 

The use case is just for proxy. So I want to make the solution simpler to just provide per client configuration.

Comment (heyitsanthony) [inline clientv3/watch.go]: Why not add an extra field into `clientv3.watchRequest` that only matters to the client? The proxy can then do `WithCreateNotify` on every watcher it creates and get the same behavior. I think regular clients might find it useful to know the start revision of particular Watchers (but not all) too. The way it is now doesn't fit well with the rest of the client design.

Comment (xiang90) [inline clientv3/watch.go]: Oh. I was not aware that the watchrequest is already saved as initReq in the stream struct. If regular watcher wants this, probably it worth the effort to do per watcher one. I will make the change.

**Comment by xiang90** [comment]
@heyitsanthony PTAL.

File: clientv3/watch.go [hunk 2]
```diff
@@ -489,19 +497,31 @@ func (w *watchGrpcStream) dispatchEvent(pbresp *pb.WatchResponse) bool {
...
 	defer w.mu.RUnlock()
 	ws, ok := w.streams[pbresp.WatchId]
+	if !ok {
+		return false
+	}
+	if pbresp.Created {
+		// ignore created event if the watcher is not requested it
+		// or we already sent the initial created event (when we are on
+		// the resume path).
+		if !ws.initReq.createdNotify || ws.lastRev != 0 {
```
Comment (heyitsanthony) [inline clientv3/watch.go]: `w.mu` doesn't protect individual `watchStream`s and `recvc` could buffer multiple create events. Probably should just be `if pbresp.Created && !ws.initReq.createdNotify { return true }` here and a `lastRev+Created` check under `<-ws.recvc` in `serveStream`.

Comment (heyitsanthony) [inline clientv3/watch.go]: On second thought, that will still queue up extra create events. If `serveStream` binds `lastRev` on the first `<-ws.recvc` when `createdNotify` is set, that should be OK.

File: clientv3/watch.go [hunk 3]
```diff
@@ -533,6 +545,14 @@ func (w *watchGrpcStream) serveStream(ws *watcherStream) {
 	for !closing {
 		curWr := emptyWr
 		outc := ws.outc
+
+		// ignore created event if create notify is not requested or
+		// we already sent the initial created event (when we are on the resume path).
+		if len(wrs) > 0 && wrs[0].Created &&
```
Comment (xiang90) [inline clientv3/watch.go]: @heyitsanthony I moved the checking here to be safe.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**